tests/model_inheritance_regress/tests.py

```diff
@@ -496,3 +496,58 @@ def test_related_filtering_query_efficiency_ticket_15844(self):
                 r.supplier_set.all(),
                 [s], lambda x: x,
             )
+
+    def test_no_query_on_parent_access(self):
+        # Refs #15250
+        italian_restaurant = ItalianRestaurant.objects.create(
+            name="Guido's House of Pasta",
+            address='944 W. Fullerton',
+            serves_hot_dogs=True,
+            serves_pizza=False,
+            serves_gnocchi=True
+        )
+
+        italian_restaurant = ItalianRestaurant.objects.get(pk=italian_restaurant.pk)
+
+        # no extra queries should be made when accessing
+        # the parent objects in a multi-table inheritance scenario.
+        with self.assertNumQueries(0):
+            restaurant = italian_restaurant.restaurant_ptr
+            place = restaurant.place_ptr
+            self.assertEqual(place.restaurant, restaurant)
+            self.assertEqual(restaurant.italianrestaurant, italian_restaurant)
+
+        italian_restaurant = (
+            ItalianRestaurant
+            .objects
+            .filter(pk=italian_restaurant.pk)
+            .only('serves_gnocchi')
+            .get()
+        )
```

Comment:
Can be simplified to `ItalianRestaurant.objects.only('serves_gnocchi').get(pk=italian_restaurant.pk)`.